src/app/store/selectors/VehicleTestResultModel.selectors.spec.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  getCurrentState,
+  getVehicleTestResultModel,
+  selectFeature,
+  selectTestTypeById
+} from '@app/store/selectors/VehicleTestResultModel.selectors';
+import { TestResultModel } from '@app/models/test-result.model';
+import { TestType } from '@app/models/test.type';
+import { select } from '@ngrx/store';
+
+describe('vehicleTestResultModel selectors', () => {
+  const createState = ({
+    vehicleTestResultModel = {
+      vehicleTestResultModel: [
+        { testTypes: [{} as TestType] } as TestResultModel,
+        { testTypes: [{} as TestType] } as TestResultModel
+      ],
+      editState: 0,
+      selectedTestResultModel: null
+    }
+  } = {}) => ({
+    vehicleTestResultModel: {
+      vehicleTestResultModel
+    }
+  });
+
+  it('selectFeature', () => {
+    const state = createState();
+    expect(selectFeature(state)).toMatchObject(state.vehicleTestResultModel);
+  });
+
+  it('getVehicleTestResultModel', () => {
+    const state = createState();
+    expect(getVehicleTestResultModel(state)).toMatchObject(
+      state.vehicleTestResultModel.vehicleTestResultModel
+    );
+  });
+
+  it('selectTestTypeById', () => {
+    const state = createState();
+    const testRes = state.vehicleTestResultModel.vehicleTestResultModel.vehicleTestResultModel;
```

Comment:
it does not seem to match the current App state we have (i don't understand why we have 3 vehicleTestResultModel nested), i am confused:

I think we have like so
```
router?: RouterReducerState;
loader: ILoaderState;
vehicleTechRecordModel: { // would rename this vehicleTechRecords
  editState: '', // this could be moved at the root also
  vehicleTechRecordModel: { // would rename this list and it will contain array of vehicleTechRecord
   // We could even have this at the root as an array of vehicleTechRecord
  }
},
vehicleTestResultModel: IVehicleTestResultModelState; // same for vehicleTestResultModel
 error?: [string] | null;
```

I think we renamed the second vehicleTestResultModel to list that will make things easier and I don't think this test is correct as it is not mapping the existing app state we have

Reply:
Refactored using projector :D